src/microdns.h

```diff
@@ -72,7 +72,8 @@ struct mdns_hdr {
         uint16_t num_add_rr;
 };
 
-typedef void (*mdns_callback)(void*, int, const struct rr_entry *);
+typedef void (*mdns_listen_callback)(void*, int, const struct rr_entry *);
+typedef void (*mdns_announce_callback)(void*, int, char *, const struct rr_entry *);
```

Comment:
The listen callback doesn't need it because it can read the IP address from the A/AAAA record (rr_entry) in the response packet.

Why store both the address string and the in_addr?

Reply:
To do like rr_entry. I'm wondering what is best way to expose addresses, the string or the struct in_addr.

Reply:
I'm happy with either, I think one should be chosen (Seems wasteful to have both when you can calculate in_addr from the address string?)

Reply:
So I would prefer keeping the struct in*_addr. That is what avahi is doing for example: https://avahi.org/doxygen/html/struct_avahi_address.html

Reply:
Ok, sounds good to me. Thanks :-)